### PR TITLE
Movie edition

### PR DESCRIPTION
**Pull request by j1nma**
## Summary
* I followed the Edición card to implement an end-point for the edition of movies, involving a MovieViewModel and UoW in MoviesRepository. Tag Helpers used in Movie Edit View.
* I made some tweaks to the MovieController methods; basically added try catch for a Db Exception I was missing from the guide on Controllers and Views.

## Trello Card Link
https://trello.com/c/to33vgIn

### Before
image

### During
image

### After. See that "1984" movie with 220 price and Drama genre no longer exists, and now is "1984 bis" with large price and "Comedy" genre.
image

### CODE REVIEW

File: src/Controllers/MoviesController.cs [hunk 1]
```diff
@@ -21,20 +21,19 @@ public IUnitOfWork UnitOfWork
         }
         public IActionResult Index()
         {
-
             var movies = _unitOfWork.MovieRepository.GetAll();
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: empty line

File: src/Controllers/MoviesController.cs [hunk 2]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
             return View();
         }
+
+        public IActionResult Edit(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 3]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(int id, MovieViewModel model)
+        {
+            if (id != model.ID)
+            {
+                return NotFound();
+            }
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 4]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+            try
+            {
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 5]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+                if (ModelState.IsValid)
+                {
+                    var movie = _unitOfWork.MovieRepository.Get(id);
+                    movie.ID = id;
+                    movie.ReleaseDate = model.ReleaseDate;
+                    movie.Genre = model.Genre;
+                    movie.Price = model.Price;
+                    movie.Title = model.Title;
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 6]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+                    _unitOfWork.MovieRepository.Update(movie);
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 7]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+                    _unitOfWork.Complete();
+                    return RedirectToAction("Index");
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 8]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+                return NotFound();
+            }
+
+            MovieViewModel model = new MovieViewModel();
+            model.ID = movie.ID;
+            model.Genre = movie.Genre;
+            model.Price = movie.Price;
+            model.ReleaseDate = movie.ReleaseDate;
+            model.Title = movie.Title;
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 9]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+            {
+                return NotFound();
+            }
+
+            var movie = _unitOfWork.MovieRepository.Get(id.Value);
+            if (movie == null)
+            {
+                return NotFound();
+            }
+
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: remove this empty line

File: src/Controllers/MoviesController.cs [hunk 10]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+            model.Price = movie.Price;
+            model.ReleaseDate = movie.ReleaseDate;
+            model.Title = movie.Title;
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Edit(int id, MovieViewModel model)
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: i don't understand why you add an 'id' param when you have it in the view model

File: src/Controllers/MoviesController.cs [hunk 11]
```diff
@@ -54,5 +53,63 @@ public IActionResult Create(MovieViewModel model)
...
+                return NotFound();
+            }
+
+            var movie = _unitOfWork.MovieRepository.Get(id.Value);
+            if (movie == null)
+            {
+                return NotFound();
+            }
+
+            MovieViewModel model = new MovieViewModel();
```
Comment (gzamudio) [inline src/Controllers/MoviesController.cs]: another way to do the same in one line and more readable:
`var model = new MovieViewModel { Id = movie.Id, Genre = movie.Genre, Price = movie.Price, ReleaseDate = movie.ReleaseDate, Title = movie.Title }`

**Review by gzamudio — CHANGES_REQUESTED** [review]

File: src/Controllers/MoviesController.cs (hunk 10 above)
Comment (j1nma) [reply src/Controllers/MoviesController.cs]: True, removed.

File: src/Views/Movies/Create.cshtml [hunk 12]
```diff
@@ -4,7 +4,7 @@
     ViewData["Title"] = "Create Movie";
 }
 
-<form asp-controller="Movies" asp-action="Create" method="post" class="form-horizontal">
+<form asp-controller="MovieViewModel" asp-action="Create" method="POST" class="form-horizontal">
```
Comment (gzamudio) [inline src/Views/Movies/Create.cshtml]: i think that 'asp-controller' must be "Movies". MovieViewModel isn't a controller

**Review by gzamudio — APPROVED** [review]

**Pull request merged**